plugins/pluginmanagers/lazy.nix

```
@@ -1,223 +1,206 @@
-
-              cmd =
-                with helpers.nixvimTypes;
-                helpers.mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on command";
-
-              ft =
-                with helpers.nixvimTypes;
-                helpers.mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on filetype";
-
-              keys =
-                with helpers.nixvimTypes;
-                helpers.mkNullOrOption (maybeRaw (either str (listOf str))) "Lazy-load on key mapping";
-
-              module = helpers.mkNullOrOption (enum [ false ]) ''
-                Do not automatically load this Lua module when it's required somewhere
-              '';
-
-              priority = helpers.mkNullOrOption number ''
-                Only useful for start plugins (lazy=false) to force loading certain plugins first.
-                Default priority is 50. It's recommended to set this to a high number for colorschemes.
-              '';
-
-              optional = helpers.defaultNullOpts.mkBool false ''
-                When a spec is tagged optional, it will only be included in the final spec,
-                when the same plugin has been specified at least once somewhere else without optional.
-                This is mainly useful for Neovim distros, to allow setting options on plugins that may/may not be part
-                of the user's plugins
-              '';
+      plugins = mkOption {
+        type = listOfPlugins;
+        default = [ ];
+        description = "List of plugins";
+      };
+    };
 
-              opts =
-                with helpers.nixvimTypes;
-                helpers.mkNullOrOption (maybeRaw (attrsOf anything)) ''
-                  opts should be a table (will be merged with parent specs),
-                  return a table (replaces parent specs) or should change a table.
-                  The table will be passed to the Plugin.config() function.
-                  Setting this value will imply Plugin.config()
-                '';
-            };
-          });
+  extraConfig =
+    cfg:
+    let
+      lazyPlugins = cfg.plugins;
 
-          listOfPlugins = types.listOf pluginType;
+      lazyPath =
+        let
+          processPlugin =
+            plugin:
+            let
+              mkEntryFromDrv =
+                p:
+                if lib.isDerivation p then
+                  {
+                    name = "${lib.getName p}";
+                    path = p;
+                  }
+                else
+                  {
+                    name = "${lib.getName p.pkg}";
+                    path = p.pkg;
+                  };
+              processDependencies =
+                if plugin ? dependencies && plugin.dependencies != null then
+                  builtins.concatMap processPlugin plugin.dependencies
+                else
+                  [ ];
+            in
+            [ (mkEntryFromDrv plugin) ] ++ processDependencies;
```

I think `mkEntryFromDrv` and `processDependencies` can probably be inlined, now that they're simplified.

```suggestion
            [ (if lib.isDerivation plugin then plugin else plugin.pkg) ]
            ++ lib.optionals (plugin.dependencies or null != null) (builtins.concatMap processPlugin plugin.dependencies);
```